src/euphorie/client/browser/training.py

```diff
@@ -238,6 +239,20 @@ def session(self):
         """Return the session for this context/request"""
         return self.context.session
 
+    def get_initial_answers(self):
+        """Pick a subset of questions, shuffle them, and initialize the answers with
+        None.
+        """
+        survey = self.webhelpers._survey
+        all_questions = survey.listFolderContents(
+            {"portal_type": "euphorie.training_question"}
+        )
+        num_training_questions = getattr(survey, "num_training_questions", None) or len(
+            all_questions
+        )
+        questions = sample(all_questions, k=num_training_questions)
+        return {q.getId(): None for q in questions}
```

Comment:
Isn't it better to get the brains from the catalog instead of the objects?

Reply:
Quicker, you mean? Yes, possibly.